### PR TITLE
Update user guide

### PR DESCRIPTION
**Pull request by jillr**
##### SUMMARY
Swap out ec2.py examples for ec2_instance. Add information about using boto
profiles and module_defaults.

##### ISSUE TYPE
- Docs Pull Request

##### COMPONENT NAME
guide_aws.rst

Depends-On: https://github.com/ansible/ansible-zuul-jobs/pull/1041
Depends-On: https://github.com/ansible/ansible-zuul-jobs/pull/1043

### CODE REVIEW

**Comment by jillr** [comment]
recheck

File: docs/docsite/rst/guide_aws.rst [hunk 1]
```diff
@@ -40,31 +40,49 @@ For environment variables::
 For storing these in a vars_file, ideally encrypted with ansible-vault::
 
     ---
-    ec2_access_key: "--REMOVED--"
-    ec2_secret_key: "--REMOVED--"
+    aws_access_key: "--REMOVED--"
+    aws_secret_key: "--REMOVED--"
```
Comment (tremble) [inline docs/docsite/rst/guide_aws.rst:44]: Should we mention `aws_session_token` ?

**Review by tremble — APPROVED** [review]
LGTM

**Comment by jillr** [comment]
recheck

**Comment by tremble** [comment]
recheck

**Comment by pabelanger** [comment]
recheck

**Comment by pabelanger** [comment]
recheck

**Comment by pabelanger** [comment]
@jillr @tremble what I would suggest here, is changes are being made to docs folder, to skip integration testing as a way to save resources.

**Comment by pabelanger** [comment]
https://github.com/ansible/ansible-zuul-jobs/pull/1043

**Comment by pabelanger** [comment]
recheck

**Comment by pabelanger** [comment]
recheck

**Pull request merged**